Description: install a CMake config file.
 This patch enables the creation of CMake's config files for
 QJson. It has been taken from upstream's GIT, taking the difference between
 tag 0.7.1 and master's commit 83c1323 of CMakeLists.txt.
 It also adds qjson-config-version.cmake.in and qjson-config.cmake.in,
 taken from the same commit mentioned above.
Author: Lisandro Damián Nicanor Pérez Meyer <perezmeyer@gmail.com>
Bug-Debian: http://bugs.debian.org/628981
Forwarded: not-needed

--- /dev/null
+++ qjson-0.7.1/qjson-config-version.cmake.in
@@ -0,0 +1,9 @@
+SET(PACKAGE_VERSION "@QJSON_LIB_VERSION_STRING@")
+IF (PACKAGE_FIND_VERSION VERSION_EQUAL PACKAGE_VERSION)
+  SET(PACKAGE_VERSION_EXACT "true")
+ENDIF (PACKAGE_FIND_VERSION VERSION_EQUAL PACKAGE_VERSION)
+IF (NOT PACKAGE_FIND_VERSION VERSION_GREATER PACKAGE_VERSION)
+  SET(PACKAGE_VERSION_COMPATIBLE "true")
+ELSE (NOT PACKAGE_FIND_VERSION VERSION_GREATER PACKAGE_VERSION)
+  SET(PACKAGE_VERSION_UNSUITABLE "true")
+ENDIF (NOT PACKAGE_FIND_VERSION VERSION_GREATER PACKAGE_VERSION)
--- /dev/null
+++ qjson-0.7.1/qjson-config.cmake.in
@@ -0,0 +1,3 @@
+SET(@CMAKE_PROJECT_NAME@_LIBRARIES "@LIB_INSTALL_DIR@/@LIB_OUT_NAME@" CACHE FILEPATH "Libraries for @CMAKE_PROJECT_NAME@")
+SET(@CMAKE_PROJECT_NAME@_INCLUDE_DIR "@INCLUDE_INSTALL_DIR@" CACHE PATH "Include path for @CMAKE_PROJECT_NAME@")
+SET(@CMAKE_PROJECT_NAME@_FOUND "TRUE")
--- qjson-0.7.1.orig/CMakeLists.txt
+++ qjson-0.7.1/CMakeLists.txt
@@ -39,7 +39,7 @@ SET( QT_DONT_USE_QTGUI TRUE )
 SET (LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)" )
 SET (LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" CACHE STRING "Directory where lib will install")
 SET (INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE PATH "The directory the headers are installed in")
-SET (CMAKE_MODULES_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/share/apps/cmake/modules" CACHE PATH "The directory to install FindQRpc.cmake to")
+SET (CMAKE_MODULES_INSTALL_DIR "${CMAKE_ROOT}/Modules" CACHE PATH "The directory to install FindQJSON.cmake to")
 
 
 set(QJSON_LIB_MAJOR_VERSION "0")
@@ -72,10 +72,24 @@ ENDIF (KDE4_BUILD_TESTS OR QJSON_BUILD_T
 add_subdirectory(cmake/modules)
 
 
-#CONFIGURE_FILE(
-#  "${CMAKE_MODULE_PATH}/cmake_uninstall.cmake.in"
-#  "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
-#  IMMEDIATE @ONLY)
+CONFIGURE_FILE(
+  "${CMAKE_MODULE_PATH}/cmake_uninstall.cmake.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
+  IMMEDIATE @ONLY)
+
+GET_TARGET_PROPERTY(LIB_OUT_NAME qjson LOCATION)
+GET_FILENAME_COMPONENT(LIB_OUT_NAME  ${LIB_OUT_NAME} NAME)
+
+# cmake-modules
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/qjson-config.cmake.in
+               ${CMAKE_CURRENT_BINARY_DIR}/qjson-config.cmake
+               @ONLY)
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/qjson-config-version.cmake.in
+               ${CMAKE_CURRENT_BINARY_DIR}/qjson-config-version.cmake
+               @ONLY)
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/qjson-config.cmake
+              ${CMAKE_CURRENT_BINARY_DIR}/qjson-config-version.cmake
+        DESTINATION "lib/cmake/qjson")
 
-#ADD_CUSTOM_TARGET(uninstall
-#  "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
+ADD_CUSTOM_TARGET(uninstall
+  "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
--- /dev/null
+++ qjson-0.7.1/cmake/modules/cmake_uninstall.cmake.in
@@ -0,0 +1,17 @@
+IF(NOT EXISTS "@CMAKE_CURRENT_BINARY_DIR@/install_manifest.txt")
+  MESSAGE(FATAL_ERROR "Cannot find install manifest: \"@CMAKE_CURRENT_BINARY_DIR@/install_manifest.txt\"")
+ENDIF(NOT EXISTS "@CMAKE_CURRENT_BINARY_DIR@/install_manifest.txt")
+
+FILE(READ "@CMAKE_CURRENT_BINARY_DIR@/install_manifest.txt" files)
+STRING(REGEX REPLACE "\n" ";" files "${files}")
+FOREACH(file ${files})
+  MESSAGE(STATUS "Uninstalling \"$ENV{DESTDIR}${file}\"")
+  EXEC_PROGRAM(
+  "@CMAKE_COMMAND@" ARGS "-E remove \"$ENV{DESTDIR}${file}\""
+  OUTPUT_VARIABLE rm_out
+  RETURN_VALUE rm_retval
+  )
+  IF(NOT "${rm_retval}" STREQUAL 0)
+  MESSAGE(FATAL_ERROR "Problem when removing \"$ENV{DESTDIR}${file}\"")
+  ENDIF(NOT "${rm_retval}" STREQUAL 0)
+ENDFOREACH(file)
